src/Service.Tests/Unittests/SqlMetadataProviderUnitTests.cs

```diff
@@ -83,6 +85,31 @@ public async Task CheckExceptionForBadConnectionStringForMsSql(string connection
             await CheckExceptionForBadConnectionStringHelperAsync(DatabaseEngine, connectionString);
         }
 
+        /// <summary>
+        /// <code>Do: </code> Tests with different combinations of connection string to ensure
+        /// tableprefix generated is correct.
+        /// <code>Check: </code> Making sure table prefix matches expected prefix.
+        /// </summary>
+        [TestMethod]
+        [DataRow("", "", "")]
+        [DataRow("", "model", "[model]")]
+        [DataRow("TestDB", "", "[TestDB]")]
+        [DataRow("TestDB", "model", "[TestDB].[model]")]
+        public void CheckTablePrefix(string databaseName, string schemaName, string expectedPrefix)
+        {
+            TestHelper.SetupDatabaseEnvironment("MSSQL");
+            RuntimeConfig baseConfigFromDisk = SqlTestHelper.SetupRuntimeConfig();
+            RuntimeConfigProvider runtimeConfigProvider = TestHelper.GenerateInMemoryRuntimeConfigProvider(baseConfigFromDisk);
+
+            ILogger<ISqlMetadataProvider> sqlMetadataLogger = new Mock<ILogger<ISqlMetadataProvider>>().Object;
+            Mock<IQueryExecutor> queryExecutor = new();
+            IQueryBuilder queryBuilder = new MsSqlQueryBuilder();
+
+            SqlMetadataProvider<SqlConnection, SqlDataAdapter, SqlCommand> provider = new MsSqlMetadataProvider(runtimeConfigProvider, queryExecutor.Object, queryBuilder, sqlMetadataLogger);
+            string tableprefix = provider.GetTablePrefix(databaseName, schemaName);
+            Assert.IsTrue(tableprefix.Equals(expectedPrefix));
```

Comment:
nit: prefer using `Assert.AreEqual`
to help easily identify what the values are in case of failures.

Reply:
fixed.